### ISSUE TITLE
Question: W1028

### ISSUE THREAD

**Issue by hanwen-pcluste** [aws-cloudformation/cfn-lint#3551]
### CloudFormation Lint Version

cfn-lint==1.9.2

### What operating system are you using?

Ubuntu in Github Checks

### Describe the bug

False positive of W1028
https://github.com/aws/aws-parallelcluster/actions/runs/10185517412/job/28175271776?pr=6372

### Expected behavior

https://github.com/aws/aws-parallelcluster/actions/runs/10185517412/job/28175271776?pr=6372 shouldn't have error

### Reproduction template

https://github.com/aws/aws-parallelcluster/blob/develop/cloudformation/networking/public-private.cfn.json

**Comment by kddejong** [comment]
For one of these you have `CreateInternetGateway` layered on itself.  You have `CreateInternetGateway` on the resource which means `CreateInternetGateway` has to be True.  Then in `GatewayId` you are using If to condition `CreateInternetGateway` and this will never be False as a result of the resource Condition.

```json
  "DefaultRouteDependsOnPublic": {
      "Condition": "CreateInternetGateway",
      "DependsOn": "VPCGatewayAttachment",
      "Properties": {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": {
          "Fn::If": [
            "CreateInternetGateway",
            {
              "Ref": "InternetGateway"
            },
            {
              "Ref": "InternetGatewayId"
            }
          ]
        },
        "RouteTableId": {
          "Ref": "RouteTablePublic"
        }
      },
      "Type": "AWS::EC2::Route"
    },
```

**Comment by kddejong** [comment]
Similar issue with this one.  The condition `ExistingInternetGateway` cannot be True at the same time `CreateInternetGateway` is True.  These two conditions are using the same parameter and empty string comparison. Its just that `ExistingInternetGateway` has a Not.

```json
"DefaultRouteNoDependsOnPublic": {
      "Condition": "ExistingInternetGateway",
      "Properties": {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": {
          "Fn::If": [
            "CreateInternetGateway",
            {
              "Ref": "InternetGateway"
            },
            {
              "Ref": "InternetGatewayId"
            }
          ]
        },
        "RouteTableId": {
          "Ref": "RouteTablePublic"
        }
      },
      "Type": "AWS::EC2::Route"
    },
```



Simplification of this would be like this.
```json
  "DefaultRouteDependsOnPublic": {
      "Condition": "CreateInternetGateway",
      "DependsOn": "VPCGatewayAttachment",
      "Properties": {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": {
          "Ref": "InternetGateway"
        },
        "RouteTableId": {
          "Ref": "RouteTablePublic"
        }
      },
      "Type": "AWS::EC2::Route"
    },
    "DefaultRouteNoDependsOnPublic": {
      "Condition": "ExistingInternetGateway",
      "Properties": {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": {
          "Ref": "InternetGatewayId"
        },
        "RouteTableId": {
          "Ref": "RouteTablePublic"
        }
      },
      "Type": "AWS::EC2::Route"
    },
```

**Comment by hanwen-pcluste** [comment]
Thank you for the information. It is indeed a problem in our code. Please feel free to close the issue
